qemu/tests/cfg/x86_cpu_flags.cfg

```diff
@@ -0,0 +1,16 @@
+- x86_cpu_flags:
+    type = x86_cpu_flags
+    cpu_model_flags += ",enforce,-mpx"
+    kill_vm_on_error = yes
+    start_vm = no
+    check_host_flags = yes
+    reboot_method = "shell"
+    only i386, x86_64
+    check_flag_cmd = "lscpu | grep Flags | awk -F ':'  '{print $2}'"
+    variants:
+        - intel:
+            vendor_id = 'intel'
+            variants:
+                - memory_protection_key:
+                    cpu_model_flags += ",+pku,check"
+                    flags = "pku ospke"
```

Comment:
@vivianQizhu @huangyum Hi, I change some struct of cfg for the later case develop, please help to check this,  thanks.

Reply:
@huangyum  some cases have nothing with vendor_id, so I suggest to current variants. How do you think?

Reply:
I see. It's ok.